sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -611,58 +545,42 @@ class SdkSynchronizer private constructor(
     override suspend fun getTransparentAddress(account: Account): String =
         processor.getTransparentAddress(account)
 
-    override fun sendToAddress(
+    @Throws(TransactionEncoderException::class)
+    override suspend fun sendToAddress(
         usk: UnifiedSpendingKey,
         amount: Zatoshi,
         toAddress: String,
         memo: String
-    ): Flow<PendingTransaction> {
-        // Using a job to ensure that even if the flow is collected multiple times, the transaction is only submitted
-        // once
-        val deferred = coroutineScope.async {
-            // Emit the placeholder transaction, then switch to monitoring the database
-            val placeHolderTx =
-                txManager.initSpend(amount, TransactionRecipient.Address(toAddress), memo, usk.account)
-
-            txManager.encode(usk, placeHolderTx).let { encodedTx ->
-                txManager.submit(encodedTx)
-            }
-
-            placeHolderTx.id
-        }
-
-        return flow<PendingTransaction> {
-            val placeHolderTxId = deferred.await()
-            emitAll(txManager.monitorById(placeHolderTxId))
+    ) {
+        txManager.encode(
+            usk,
+            amount,
+            TransactionRecipient.Address(toAddress),
+            memo,
+            usk.account
+        ).let { encodedTx ->
+            txManager.submit(encodedTx)
         }
     }
 
-    override fun shieldFunds(
+    @Throws(TransactionEncoderException::class)
+    override suspend fun shieldFunds(
         usk: UnifiedSpendingKey,
         memo: String
-    ): Flow<PendingTransaction> {
+    ) {
```

Comment:
We should return a transaction data type for wallets to show or look up is needed.